### PR TITLE
qa/test_full: always catch the CommandFailedError when the disk is full

### PR DESCRIPTION
**Pull request by lxbsz**
Fixes: https://tracker.ceph.com/issues/45145
Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/tasks/cephfs/test_full.py [hunk 1]
```diff
@@ -132,16 +132,21 @@ def _test_full(self, easy_case):
 
         # Fill up the cluster.  This dd may or may not fail, as it depends on
         # how soon the cluster recognises its own fullness
-        self.mount_a.write_n_mb("large_file_a", self.fill_mb // 2)
         try:
-            self.mount_a.write_n_mb("large_file_b", self.fill_mb // 2)
+            self.mount_a.write_n_mb("large_file_a", self.fill_mb // 2)
         except CommandFailedError:
             log.info("Writing file B failed (full status happened already)")
```
Comment (gregsfortytwo) [inline qa/tasks/cephfs/test_full.py:138]: s/B/A/

**Comment by gregsfortytwo** [comment]
Rest looks fine though

File: qa/tasks/cephfs/test_full.py (hunk 1 above)
Comment (lxbsz) [reply qa/tasks/cephfs/test_full.py:138]: Will fix it.

**Comment by lxbsz** [comment]
The integration test passed: http://pulpito.ceph.com/xiubli-2020-04-24_06:48:53-fs:basic_functional-wip-fs-misc-2020-04-24-1310-distro-basic-smithi/4979752/

**Comment by batrick** [comment]
Also, I think this is a recent issue because it looks like recent runs of Octopus and NAutilus exhibit this problem: http://pulpito.ceph.com/teuthology-2020-05-04_04:10:02-fs-octopus-distro-basic-smithi/5020440/

**Comment by lxbsz** [comment]
> @lxbsz please do not forget to include the path to the teuthology log in your bug report: https://tracker.ceph.com/issues/45145
> 
Sure, will do it in future.

> This fix is not correct: the first write should succeed otherwise our computed value of `self.fill_mb` must be wrong. After all, we're only writing half of what we think will fill the cluster. I would look in the logs to see why the `fill_mb` value is incorrect.

As I remembered I could also reproduce it locally by vstart_runner.py to run the whole `TestClusterFull` task.

**Comment by batrick** [comment]
> > This fix is not correct: the first write should succeed otherwise our computed value of `self.fill_mb` must be wrong. After all, we're only writing half of what we think will fill the cluster. I would look in the logs to see why the `fill_mb` value is incorrect.
> 
> As I remembered I could also reproduce it locally by vstart_runner.py to run the whole `TestClusterFull` task.

Okay, but why is the cluster full before a finishes writing? That's the question!

**Comment by lxbsz** [comment]
> > > This fix is not correct: the first write should succeed otherwise our computed value of `self.fill_mb` must be wrong. After all, we're only writing half of what we think will fill the cluster. I would look in the logs to see why the `fill_mb` value is incorrect.
> > 
> > 
> > As I remembered I could also reproduce it locally by vstart_runner.py to run the whole `TestClusterFull` task.
> 
> Okay, but why is the cluster full before a finishes writing? That's the question!

It seems the previous test case didn't clean it or something else. I will check and try to reproduce it again to find out whether there has any bug in qa or the ceph.

**Comment by lxbsz** [comment]
> > > > This fix is not correct: the first write should succeed otherwise our computed value of `self.fill_mb` must be wrong. After all, we're only writing half of what we think will fill the cluster. I would look in the logs to see why the `fill_mb` value is incorrect.
> > > 
> > > 
> > > As I remembered I could also reproduce it locally by vstart_runner.py to run the whole `TestClusterFull` task.
> > 
> > 
> > Okay, but why is the cluster full before a finishes writing? That's the question!
> 
> It seems the previous test case didn't clean it or something else. I will check and try to reproduce it again to find out whether there has any bug in qa or the ceph.

I had tried to reproduce it for many times, but not successed, I will check it again later.

**Comment by batrick** [comment]
@lxbsz please revive/rebase this changeset in a new PR.

**Comment by lxbsz** [comment]
Create a new PR: https://github.com/ceph/ceph/pull/40717.
Closing this one.

**Pull request closed without merging**